assets/js/atomic/blocks/product-elements/button/supports.ts

```diff
@@ -32,4 +31,9 @@ export const supports = {
 		__experimentalSelector:
 			'.wp-block-button.wc-block-components-product-button .wc-block-components-product-button__button',
 	} ),
+	...( typeof __experimentalGetSpacingClassesAndStyles === 'function' && {
+		spacing: {
+			margin: true,
+		},
+	} ),
```

Comment:
Heads-up that this overrides the `padding` attribute set in L22.

I think we can also enable padding in WC core. It should be fine because `getSpacingClassesAndStyles` [has graduated](https://github.com/WordPress/gutenberg/blob/e9e62d3a10d57b581ca88ca89ee46a88d5d8507d/packages/block-editor/src/hooks/use-spacing-props.js#L18).


```suggestion
	...( typeof __experimentalGetSpacingClassesAndStyles === 'function' && {
		spacing: {
			margin: true,
			padding: true,
		},
	} ),
```

I didn't test it, but I think it should work.

Reply:
Ouch, disregard my previous comment, `padding` needs `__experimentalSkipSerialization`, which I don't think has graduated yet. :thinking: